source/common/network/io_socket_handle_impl.cc

```diff
@@ -16,6 +18,48 @@ using Envoy::Api::SysCallSizeResult;
 namespace Envoy {
 namespace Network {
 
+namespace {
+
+void getSocketAddressInfo(const Address::Instance& address, sockaddr_storage& addr, socklen_t& sz) {
+  // TODO(sumukhs): Add support for unix domain sockets
+
+  const Address::Ip* ip = address.ip();
+  ASSERT(ip != nullptr);
+  memset(&addr, 0, sizeof(addr));
+
+  switch (ip->version()) {
+  case Address::IpVersion::v4: {
+    addr.ss_family = AF_INET;
+    auto const* ipv4 = ip->ipv4();
+    ASSERT(ipv4 != nullptr);
+    sockaddr_in* addrv4 = reinterpret_cast<sockaddr_in*>(&addr);
+    addrv4->sin_port = htons(ip->port());
+    addrv4->sin_addr.s_addr = ipv4->address();
+    sz = sizeof(sockaddr_in);
+    break;
+  }
+  case Address::IpVersion::v6: {
+    addr.ss_family = AF_INET6;
+    auto const* ipv6 = ip->ipv6();
+    ASSERT(ipv6 != nullptr);
+    sockaddr_in6* addrv6 = reinterpret_cast<sockaddr_in6*>(&addr);
+    addrv6->sin6_port = htons(ip->port());
+
+    const auto address = ipv6->address();
+    memcpy(static_cast<void*>(&addrv6->sin6_addr.s6_addr), static_cast<const void*>(&address),
+           sizeof(absl::uint128));
+
+    sz = sizeof(sockaddr_in6);
+    break;
+  }
+  default:
+    NOT_REACHED_GCOVR_EXCL_LINE;
+    break;
```

Comment:
nit: shouldn't need break here, it will never execute.

Reply:
ack